images/nginx-drupal/drupal.conf

```diff
@@ -7,50 +7,61 @@ server {
   include /etc/nginx/helpers/*.conf;
 
   root /app/${WEBROOT:-};
-  index  index.php;
-
-  ## rewriting /index.php to / because after https://www.drupal.org/node/2599326
-  ## autocomplete URLs are forced to go to index.php
-  rewrite ^/index.php / last;
+  index index.php;
 
   ## The 'default' location.
   location / {
     include /etc/nginx/conf.d/drupal/location_prepend*.conf;
 
     ## Do not allow access to .txt and .md unless inside sites/*/files/
-    location ~* ^(?!.+sites\/.+\/files\/).+\.(txt|md)$ {
+    location ~* ^(?!.+sites\/.+\/files\/).+\.(txt|md|log)$ {
       deny all;
       access_log off;
       log_not_found off;
+      return 404;
     }
 
     ## Replicate the Apache <FilesMatch> directive of Drupal standard
     ## .htaccess. Disable access to any code files. Return a 404 to curtail
     ## information disclosure.
-    location ~* \.(engine|inc|install|make|module|profile|po|sh|.*sql|theme|twig|tpl(\.php)?|xtmpl|yml)(~|\.sw[op]|\.bak|\.orig|\.save)?$|^\/(\.(?!well-known).*|Entries.*|Repository|Root|Tag|Template|composer\.(json|lock))$|^\/#.*#$|\.php(~|\.sw[op]|\.bak|\.orig|\.save)$ {
+    location ~* \.(engine|inc|install|make|module|profile|po|sh|.*sql|theme|twig|tpl(\.php)?|xtmpl|yml)(~|\.sw[op]|\.bak|\.orig|\.save)?$|composer\.(lock|json)$|web\.config$|^(\.(?!well-known).*|Entries.*|Repository|Root|Tag|Template)$|^#.*#$|\.php(~|\.sw[op]|\.bak|\.orig|\.save)$ {
       deny all;
       access_log off;
       log_not_found off;
+      return 404;
     }
 
     ## Expiring per default for four weeks and one second, Drupal will overwrite that if necessary
     expires ${NGINX_DEFAULT_EXPIRES:-2628001s};
 
-    ## Disallow access to any dot files, but send the request to Drupal
-    location ~* /\. {
-      try_files /dev/null @drupal;
```

Comment:
removing this will cause issues. Drupal allows URLs that start with a dot: http://nginx.master.drupal-example-9.test1.amazee.io/foobar/.dotfilesforethewin
your proposed regex: `location ~ (^|/)\. {` would match this and block traffic to this.